app/src/main/java/app/revanced/integrations/patches/playback/speed/RememberPlaybackSpeedPatch.java

```diff
@@ -0,0 +1,82 @@
+package app.revanced.integrations.patches.playback.speed;
+
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import app.revanced.integrations.settings.SettingsEnum;
+import app.revanced.integrations.utils.LogHelper;
+import app.revanced.integrations.utils.ReVancedUtils;
+
+public final class RememberPlaybackSpeedPatch {
+
+    /**
+     * The current playback speed
+     */
+    private static float currentPlaybackSpeed = getLastRememberedPlaybackSpeed();
+
+    private final static float DEFAULT_PLAYBACK_SPEED = (float) SettingsEnum.REMEMBER_PLAYBACK_SPEED_LAST_SELECTED_VALUE.getDefaultValue();
+
+    @Nullable
+    private static String currentVideoId;
+
+
+    private static void showToast(final String message) {
+        Toast.makeText(ReVancedUtils.getContext(), message, Toast.LENGTH_LONG).show();
+    }
+
+    private static float getLastRememberedPlaybackSpeed() {
+        return SettingsEnum.REMEMBER_PLAYBACK_SPEED_LAST_SELECTED_VALUE.getFloat();
+    }
+
+    private static void rememberPlaybackSpeed() {
+        SettingsEnum.REMEMBER_PLAYBACK_SPEED_LAST_SELECTED_VALUE.saveValue(currentPlaybackSpeed);
+    }
+
+    private static boolean rememberLastSelectedPlaybackSpeed() {
+        return SettingsEnum.REMEMBER_PLAYBACK_SPEED_LAST_SELECTED.getBoolean();
+    }
+
+    /**
+     * Injection point.
+     * Called when a new video loads.
+     */
+    public static void newVideoLoaded(@NonNull String videoId) {
+        if (videoId.equals(currentVideoId)) {
+            return;
+        }
+        currentVideoId = videoId;
+        currentPlaybackSpeed = getLastRememberedPlaybackSpeed();
+    }
```

Comment:
Instead of this entire method and injection point `getLastRememberedPlaybackSpeed();` can be returned by the injection point `getCurrentPlaybackSpeed` or am I missing something.